### PR TITLE
Refactor test names and remove flaky test

### PR DESCRIPTION
**Pull request by Olavhaasie**
Closes PM-138

## Description
I removed the flaky test, since there was no way in fixing this in a consistent way. I think that the other tests are also flaky, so they might fail now, but we'll see. I also refactored the names of the tests and removed useless stuff.

## Changes
* Removed `belowSpecificThreshold`
* Renamed tests
* Fix types

### CODE REVIEW

File: march_safety/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(march_safety)
 
-add_compile_options(-std=c++11 -Wall -g)
+add_compile_options(-std=c++11 -Wall -Wextra)
```
Comment (Roelemans) [inline march_safety/CMakeLists.txt]: What does this do?

File: march_safety/CMakeLists.txt [hunk 2]
```diff
@@ -71,31 +71,30 @@ if (CATKIN_ENABLE_TESTING)
...
         test/${PROJECT_NAME}_connection_lost.test
-        test/TestConnectionLost.cpp
+        test/test_connection_lost.cpp
     )
 
-    add_rostest_gtest(${PROJECT_NAME}-test-connection-never-started
+    add_rostest_gtest(${PROJECT_NAME}_connection_never_started_test
         test/${PROJECT_NAME}_connection_never_started.test
-        test/TestConnectionNeverStarted.cpp
+        test/test_connection_never_started.cpp
```
Comment (Roelemans) [inline march_safety/CMakeLists.txt]:
```suggestion
        test/connection_never_started_test.cpp
```

File: march_safety/CMakeLists.txt [hunk 3]
```diff
@@ -71,31 +71,30 @@ if (CATKIN_ENABLE_TESTING)
...
     )
 
-    add_rostest_gtest(${PROJECT_NAME}-test-connection-not-lost
+    add_rostest_gtest(${PROJECT_NAME}_connection_not_lost_test
         test/${PROJECT_NAME}_connection_not_lost.test
-        test/TestConnectionNotLost.cpp
+        test/test_connection_not_lost.cpp
```
Comment (Roelemans) [inline march_safety/CMakeLists.txt]:
```suggestion
        test/connection_not_lost_test.cpp
```

File: march_safety/CMakeLists.txt (hunk 1 above)
Comment (Olavhaasie) [reply march_safety/CMakeLists.txt]: Adds extra warnings to compilation

**Pull request merged**